url9a01f43c15_poragchowdhury_securitygame_tgz-pJ8-securitygame_GameMasterJ8/src/securitygame/Analyzer.java

```diff
@@ -0,0 +1,298 @@
+package securitygame;
+
+import java.util.Arrays;
+
+/**
+ * Calculates stats and performance of Defenders and Attackers
+ * @author Oscar Veliz
+ * @version 2014/11/07
+ */
+public class Analyzer
+{
+    int[][] points;
+    String[] defenderNames;
+    String[] attackerNames;
+    int numAttackers;
+    int numDefenders;
+    int[] aTotals;
+    int[] dTotals;
+    double[] aAverage;
+    double[] dAverage;
+    double[] aMedian;
+    double[] dMedian;
+    double[] aStd;
+    double[] dStd;
+    double[] aRegret;
+    double[] dRegret;
+    double[] aBestOf;
+    double[] dBestOf;
+
+    /**
+     * Calculates important values for comparing attackers and defenders
+     * @param results point value 2dArray with row as defenders and columns as attackers
+     * @param aNames the names of the attackers as they are in the results
+     * @param dNames the names of the defenders as they are in the results
+     */
+    public Analyzer(int[][] results, String[] aNames, String[] dNames)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 18. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 90-116 into a separate method.